model/src/main/kotlin/config/ScannerCompatibilityConfiguration.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.config
+
+/**
+ * A data class storing configuration options related to the compatibility of scan results.
+ *
+ * The options defined in this class are evaluated when scan results are looked up from a scan results storage. It
+ * then has to be checked whether the results from the storage has been produced by a compatible scanner version with
+ * compatible command line options. Per default, this check is rather strict. By specifying some of these options of
+ * this class, it can be configured that scan results are reused even if their parameters do not exactly match the
+ * current scanner settings.
+ */
+data class ScannerCompatibilityConfiguration(
```

Comment:
A general remark about the class docs and name: IMO it's a bit misleading to refer to the (currently) only place where this class is used, i.e. as part of the scanner configuration, for compatibility checks. I'd prefer to look at this class a bit more in isolation, and at the plain properties it stores. To me, this is something like a `ToolVersionMatcher`; it's not even scanner-specific! We could also make use of it in the Analyzer when checking CLI tools it calls. I believe the docs and class name should reflect that, although one could probably come up with a better class name than the one I mentioned.

Reply:
You are right in the current state of the implementation. However, the class is going to be extended with additional properties that make it specific to the use case described in the docs: namely a set of patterns determining the scanner options to ignore when checking the compatibility of scan results.

It may be possible to refactor some functionality out if a generic tool version check is needed at multiple places. This could be done in a later step. (It sounds that this would affect a larger number of classes.) Also note that in its current state, the class is a plain data class and does not contain any matching logic. Therefore, a name ending on `Matcher` would be misleading.

Reply:
> namely a set of patterns determining the scanner options to ignore when checking the compatibility of scan results.

These "scanner options" are just command line options, correct? In that case I'd still argue that even with that addition, this very same class could be used e.g. to configure the compatibility of command line tools used by the analyzer, or?

> Therefore, a name ending on `Matcher` would be misleading.

Agreed, whatever the name would be it should probably not include "Matcher".

Reply:
> These "scanner options" are just command line options, correct? In that case I'd still argue that even with that addition, this very same class could be used e.g. to configure the compatibility of command line tools used by the analyzer, or?
> 

Well, if the use case is that generic. So far I have seen the compatibility checks as a scanner-specific issue. What is a bit tricky here is that the class actually does not define the compatibility of a specific tool, but of a result produced by a specific tool that was invoked with specific parameters. Is the situation similar in the analyzer use case? It seems to me that here only the compatibility of the tool itself is relevant; if a tool is considered compatible, its results (the set of dependencies in this case) are always accepted.